### ISSUE TITLE
Missing items on facet pages from record app "view more" links

### ISSUE THREAD

**Issue by ljpearlman** [informatics-isi-edu/chaise#1587]
In some cases, clicking "View More" on a record page will pop up a faceted search that has no indication of what filters have been applied. For example, going to https://dev.rebuildingakidney.org/chaise/record/#2/Common:Gene/RID=Q-4854 and clicking "view more" by the "Antibody Gene Association" section will take you to https://dev.rebuildingakidney.org/chaise/recordset/#2/Antibody:Antibody_Gene_Association/*::facets::N4IghgdgJiBcDaoDOB7ArgJwMYFM6JHQBcAjdafEEAGhAEEIiBLMqATwH0BxHCHDuklRYmYZiggcAcgGEAQgEluvHAoAiHAGYBrHGxABdAL60ASusO0sACxRNcSSgEUAtABYAHAFY3ho8aA@sort(RID),
which is (correctly) filtered on the appropriate gene value, but which has no pill to that effect at the top of the page and has no "Gene" facet on the left-hand side.

This only happens on some tables, not others. This may be a red herring, but so far all the broken ones I've found hang off the Gene table, and the foreign keys involved all refer to the integer-valued column NCBI_GeneID.

**Comment by RFSH** [comment]
There might be several reasons that you’re not seeing the applied facets. Current implementation does not handle having choice and range (or any combination) of facet types together. It will only show filters of one type.

This case is related to the logic that we added for handling alternative tables (which we might want to remove or get back to and change). The table Common:Gene has an alternative for compact (Gene_Thumbnail_Display). The facet definition is based on `Gene` table but since it’s entity mode we are going to show that table in `compact/select` context, which means showing the `Gene_Thumbnail_Display` table. Therefore in ermrestjs we’re actually making sure the facet definition is pointing to the correct table. This will ensure all the filters are being applied properly. While doing so, we might end up removing the facet definition from the list (because we cannot handle it), which is what is happening here. It’s this case: https://github.com/informatics-isi-edu/ermrestjs/wiki/Table-Alternatives#scenario-1-entity-facet-with-alternative-table-for-compactselect

Now since we’re completely removing that facet from the list, we cannot show the applied filters for it either.

**Comment by RFSH** [comment]
To expand on my explanation, this is what's happening:

- You click on view more in the record app for Common:Gene/RID=Q-4854.
- Chaise will redirect you to the Gene_Expression:Specimen recordset app with the following facet:
  ```
  {
    "and": [
      {"source": [{"inbound": fk1}, {"outbound": fk2} ,"RID"], "choices": ["Q-4854"]}
    ]
  }
  ```
  (fk1 and fk2 are the foreignkeys that are used for the association table).
- While trying to process the list of facets, ERMrestJS will encounter that facet definition to show entity picker for table `Gene`. And since it's entity picker, we're using `compact/select` context for showing the table. And `Gene` has an alternative table defined for `compact` context. So we need to show that table instead. Therefore we need to make sure that ermrestjs/chaise understands the relationship between the base (`Specimen`) table and the table that we're showing in facet (`Gene_Thumbnail_Display`). 
- Since the facet definition is based on `RID`, and relationship between alternative table and base table is based on another column, ermrestjs cannot retrace this path and completely ignores it. This is happening because in entity mode, ermrestjs will build the filters based on the key column that you have defined on your face definition and we cannot easily map the key column of `Gene_Thumbnail_Display` table to `RID` column of `Gene` table.


TL;DR to fix this, you need to make sure the key column that you're using for alternative table is the RID of Gene table (The key used for alternative table should be the same key used for faceting). Then we should be able to show the appropriate facet (`Gene_Thumbnail_Display` table) on the left panel.

**Comment by ljpearlman** [comment]
Just to be clear, every incoming foreign key for the Gene table uses the NCBI_GeneID column. Are you saying that I just need to change the foreign key relationship of the alternative table, or do I need to refactor all the tables?

**Comment by RFSH** [comment]
Just the foreignkey for the alternative table. changing the foreignkey relationship to be based on RID (the same key used for facet definition), should resolve this issue.

**Comment by RFSH** [comment]
I am going to close this issue since it is a case that we said we're not handling. Besides at some point we should look back at the alternative table feature and fix these sort of issues with it.
